lib/libspl/asm-generic/atomic.c

```diff
@@ -230,24 +192,14 @@ ATOMIC_ADD_NV(64, uint64_t, int64_t)
 void *
 atomic_add_ptr_nv(volatile void *target, ssize_t bits)
 {
-	void *ptr;
-
-	VERIFY3S(pthread_mutex_lock(&atomic_lock), ==, 0);
-	ptr = (*(caddr_t *)target += bits);
-	VERIFY3S(pthread_mutex_unlock(&atomic_lock), ==, 0);
-
-	return (ptr);
+	return __atomic_add_fetch((void **)target, bits, __ATOMIC_SEQ_CST);
```

Comment:
cstyle: missing parenthesis around return value.  I'm surprised the style checker didn't flag this.

Reply:
almost the whole file is BEGIN CSTYLEDed, so probably that

Reply:
yeah it is that: https://github.com/openzfs/zfs/pull/11904/checks?check_run_id=2364969990#step:5:6